util/kubeconfig/kubeconfig.go

```diff
@@ -44,8 +44,12 @@ var (
 )
 
 // FromSecret fetches the Kubeconfig for a Cluster.
-func FromSecret(ctx context.Context, c client.Reader, cluster client.ObjectKey) ([]byte, error) {
-	out, err := secret.Get(ctx, c, cluster, secret.Kubeconfig)
+func FromSecret(ctx context.Context, c client.Reader, cluster client.ObjectKey, userKubeconfig bool) ([]byte, error) {
```

Comment:
Changing the function signature here will cause a breaking API code change.

Reply:
@valaparthvi could you introduce a new function? We can then refactor the existing one to call the new one.

Reply:
same as above, (what about having `FromUserSecret`)

Reply:
I can make that work.